File: crates/sqlite/src/queries/entries.rs
use colette_database::entries::InsertData;
use sqlx::{Error, SqliteExecutor};

pub async fn insert(ex: impl SqliteExecutor<'_>, data: InsertData<'_>) -> Result<i64, Error> {
    let row = sqlx::query_file!(
        "queries/entries/insert.sql",
        data.link,
        data.title,
        data.published_at,
        data.description,
        data.author,
        data.thumbnail_url
    )
    .fetch_one(ex)
    .await?;

    Ok(row.id)
}

File: ts/packages/postgres/src/queries/entries.ts
import { sql } from 'drizzle-orm'
import type { Database } from '../client'
import { entriesTable } from '../schema'
import type { EntryInsert } from '../types'

export async function insertEntry(db: Database, data: EntryInsert) {
	return db
		.insert(entriesTable)
		.values(data)
		.onConflictDoUpdate({
			target: entriesTable.link,
			set: {
				title: sql`excluded.title`,
				publishedAt: sql`excluded.published_at`,
				description: sql`excluded.description`,
				author: sql`excluded.author`,
				thumbnailUrl: sql`excluded.thumbnail_url`,
			},
		})
		.returning({
			id: entriesTable.id,
		})
}
